tests/openedx_tagging/core/tagging/test_views.py

```diff
@@ -0,0 +1,299 @@
+        (True, status.HTTP_200_OK),
+        (False, status.HTTP_200_OK),
+        ("True", status.HTTP_200_OK),
+        ("False", status.HTTP_200_OK),
+        ("1", status.HTTP_200_OK),
+        ("0", status.HTTP_200_OK),
+        (2, status.HTTP_400_BAD_REQUEST),
+        ("invalid", status.HTTP_400_BAD_REQUEST),
+    )
+    @ddt.unpack
+    def test_list_taxonomy_queryparams(self, enabled, expected_status):
+        url = reverse("oel_tagging:taxonomy-list")
+        Taxonomy.objects.create(name="Taxonomy enabled 1", enabled=True).save()
+        Taxonomy.objects.create(name="Taxonomy enabled 2", enabled=True).save()
+        Taxonomy.objects.create(name="Taxonomy disabled", enabled=False).save()
+
+        self.client.force_authenticate(user=self.staff)
+        if enabled is not None:
+            response = self.client.get(url, {"enabled": enabled})
+        else:
+            response = self.client.get(url)
+        self.assertEqual(response.status_code, expected_status)
+
+        # If we were able to create the taxonomy, check if it was created
+        if status.is_success(expected_status):
+            if enabled is None:
+                self.assertEqual(len(response.data), 3)
+            elif enabled in [True, "True", 1, "1"]:
+                self.assertEqual(len(response.data), 2)
+            else:
+                self.assertEqual(len(response.data), 1)
+
+
+    @ddt.data(
+        (None, status.HTTP_403_FORBIDDEN),
+        ("user", status.HTTP_403_FORBIDDEN),
+        ("staff", status.HTTP_200_OK),
+    )
+    @ddt.unpack
+    def test_list_taxonomy(self, user_attr, expected_status):
+        url = reverse("oel_tagging:taxonomy-list")
+
+        user = getattr(self, user_attr) if user_attr else None
+        if user:
+            self.client.force_authenticate(user=user)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, expected_status)
+
+    @ddt.data(
+        (None, {"enabled": True}, status.HTTP_403_FORBIDDEN),
+        ("user", {"enabled": True}, status.HTTP_200_OK),
+        ("user", {"enabled": False}, status.HTTP_404_NOT_FOUND),
+        ("staff", {"enabled": True}, status.HTTP_200_OK),
+        ("staff", {"enabled": True}, status.HTTP_200_OK),
+    )
+    @ddt.unpack
+    def test_detail_taxonomy(self, user_attr, taxonomy_data, expected_status):
+        create_data = {**{"name": "taxonomy detail test"}, **taxonomy_data}
+        taxonomy = Taxonomy.objects.create(**create_data)
+        url = reverse("oel_tagging:taxonomy-detail", kwargs={"id": taxonomy.id})
+
+        user = getattr(self, user_attr) if user_attr else None
+        if user:
+            self.client.force_authenticate(user=user)
+
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, expected_status)
+
+        if status.is_success(expected_status):
+            response = self.client.get(url)
+            self.assertGreaterEqual(response.data.items(), create_data.items())
```

Comment:
I'm confused.. why do we need to call `get` again?
And instead of just checking that the data fields are the right length, could you check the returned data itself? We can add a helper method for this. (Applies everywhere there's an `assertGreaterEqual`)

```suggestion
        if status.is_success(expected_status):
            self.check_taxonomy(response.data, taxonomy.id, **create_data)
            
    def check_taxonomy(self, data, id, name="", description=None, enabled=True, required=False, allow_multiple=False, allow_free_text=False):
        """
        Check that the data dict contains the expected fields.
        """
        assert data['id'] == id
        assert data['name'] == name
        assert data['enabled'] == enabled
        assert data['description'] == description
        assert data['required'] == required
        assert data['allow_multiple'] == allow_multiple
        assert data['allow_free_text'] == allow_free_text
        assert data['system_defined'] == False
```

Reply:
My bad here! We don't need to call get twice!

But the `assertGreaterEqual` in that way doesn't compare the count. It compares if the second dict is a subset from the first (the data itself).

```python
a = {
  "foo": "foo",
  "bar": "bar",
}
b = {
  "foo": "foo",
}
c = {
  "foo": "fzz",
}

self.assertGreaterEqual(a.items(), b.items()) # This pass
self.assertGreaterEqual(a.items(), c.items()) # This FAILS
# The order matters
self.assertGreaterEqual(b.items(), a.items()) # This FAILS
```

Do you prefer the explicity check?

Reply:
Ah cool, I wasn't aware that this was how assertGreaterEqual worked with dicts!

But yes please, I think it's clearer to verify all the fields on the taxonomy are as expected.